src/wpantund/NCPInstanceBase-Addresses.cpp

```diff
@@ -918,38 +918,80 @@ NCPInstanceBase::find_prefix_entry(const IPv6Prefix &prefix, const OnMeshPrefixE
 	return iter;
 }
 
+std::multimap<NCPInstanceBase::IPv6Prefix, NCPInstanceBase::OnMeshPrefixEntry>::iterator
+NCPInstanceBase::find_domain_prefix_entry(const IPv6Prefix &prefix, const OnMeshPrefixEntry &entry)
+{
+	std::multimap<IPv6Prefix, OnMeshPrefixEntry>::iterator iter;
+
+	iter = mOnMeshPrefixes.begin();
+
+	for (; iter != mOnMeshPrefixes.end(); ++iter)
+	{
+		if (((uint16_t)((*iter).second.get_flags()) & OnMeshPrefixEntry::kFlagExtDP) == OnMeshPrefixEntry::kFlagExtDP)
+		{
+			break;
+		}
+	}
+
+	return iter;
+}
+
+
 void
 NCPInstanceBase::on_mesh_prefix_was_added(Origin origin, const struct in6_addr &prefix_address, uint8_t prefix_len,
 	uint16_t flags, bool stable, uint16_t rloc16, CallbackWithStatus cb)
 {
 	IPv6Prefix prefix(prefix_address, prefix_len);
 	OnMeshPrefixEntry entry(origin, flags, stable, rloc16);
 	std::multimap<IPv6Prefix, OnMeshPrefixEntry>::iterator iter;
+	bool add_new_domain_entry = false;
 
-	iter = find_prefix_entry(prefix, entry);
+	if ((flags & OnMeshPrefixEntry::kFlagExtDP) == OnMeshPrefixEntry::kFlagExtDP)
+	{
+		iter = find_domain_prefix_entry(prefix, entry);
 
-	if (iter == mOnMeshPrefixes.end()) {
+		if (iter != mOnMeshPrefixes.end())
+		{
+			syslog(LOG_INFO, "OnMeshPrefixes: Removing old domain prefix %s", entry.get_description(prefix).c_str());
+			mOnMeshPrefixes.erase(iter);
+		}
+		add_new_domain_entry = true;
+	}
+	else
+	{
+		iter = find_prefix_entry(prefix, entry);
+		if (iter == mOnMeshPrefixes.end())
+		{
+			add_new_domain_entry = true;
+		}
+	}
+
+	if (add_new_domain_entry)
+	{
 		mOnMeshPrefixes.insert(std::make_pair(prefix, entry));
 		syslog(LOG_INFO, "OnMeshPrefixes: Adding %s", entry.get_description(prefix).c_str());
 
-		if (origin != kOriginThreadNCP) {
+		if (origin != kOriginThreadNCP)
+		{
```

Comment:
`wpantund` traditionally followed the style where `{` starts on the same line. 
We don't have or enforce any specific style guide for it though. 
I would suggest no need to change existing code. New code we can be more flexible.